src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIBackButtonXamlUpdater.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    [ApplicableComponents(ProjectComponents.WinUI)]
+    internal class WinUIBackButtonXamlUpdater : IUpdater<IProject>
+    {
+        public const string NewBackButtonName = "UAGeneratedBackButton";
+
+        public const string RuleID = "UA308";
+
+        public string Id => typeof(WinUIBackButtonXamlUpdater).FullName;
+
+        public string Title => "Insert back button in XAML";
+
+        public string Description => "Adds a back button to all WinUI XAML pages.";
+
+        private const string AnalysisString = "Default back button in the title bar does not exist in WinUI3 apps. You have to add your own back button.";
+
+        public BuildBreakRisk Risk => BuildBreakRisk.Medium;
+
+        public async Task<IUpdaterResult> ApplyAsync(IUpgradeContext context, ImmutableArray<IProject> inputs, CancellationToken token)
+        {
+            await Task.Yield();
+            foreach (var project in inputs)
+            {
+                foreach (var file in project.FindFiles(".xaml", ProjectItemType.None))
+                {
+                    var doc = new XmlDocument();
+                    doc.Load(file);
+                    Dictionary<string, string> xmlNamespaces = new Dictionary<string, string>()
```

Comment:
can this be made once? Or is a new instance needed for every file in every project?